pysingfel/detector/base.py

```diff
@@ -246,9 +270,59 @@ def get_photons(self, particle, device=None):
                 "Device option is deprecated. "
                 "Everything now runs on the GPU.")
 
-        raw_data = self.get_pattern_without_corrections(particle=particle)
+        raw_data = self.get_pattern_without_corrections(particle=particle,return_type="intensity")
         return self.add_correction_and_quantization(raw_data)
 
+
+    def maxRadius(self, particles):
+        radius_current = 0
+        for particle in particles:
+            print (particle, '->', particles[particle])
+            radius_arr = particle.atom_pos - np.mean(particle.atom_pos, axis=0)
+            for row in radius_arr:
+                radius = np.sqrt(row[0]**2+row[1]**2+row[2]**2)
+                if radius > radius_current:
+                    radius_current = radius
+        radius_max = radius_current
+        return radius_max
+
+
+    def distribute(self, particles, beam_focus_radius, jet_radius): #beam_focus_radius = 10e-6 #jet_radius = 1e-4
+        state = []
+        for particle in particles:
+            for count in range(particles[particle]):
+                state.append(particle)
+        radius_max = self.maxRadius(particles)
+        N = sum(particles.values()) # total number of particles
+        coords = np.zeros((N,3)) # initialize N*3 array
+        # generate N*3 random positions
+        for i in range(N):
+            coords[i,0] = beam_focus_radius*np.random.uniform(-1, 1)
+            coords[i,1] = beam_focus_radius*np.random.uniform(-1, 1)
+            coords[i,2] = jet_radius*np.random.uniform(-1, 1)
+        # calculate N*N distance matrix
+        dist_matrix = distance.cdist(coords, coords, 'euclidean')
+        # collision detection check (<2 maxRadius)
+        collision = dist_matrix < 2*radius_max
+        checkList = [collision[i][j] for i in range(N) for j in range(N) if j > i]
+        if any(item == True for item in checkList):
```

Comment:
I thought we discussed removing any colliding particles. This recursive function looks like it's trying to fit the exact number of particles?